### PR TITLE
Feature login changes

### PR DESCRIPTION
**Pull request by Fasse**
#580

### CODE REVIEW

**Comment by Fasse** [comment]
Can someone please review?

File: adm_program/system/classes/session.php [hunk 1]
```diff
@@ -459,19 +402,58 @@ public function setAutoLogin()
...
+     *                         With "0" the cookie will expire if the session ends. (When Browser gets closed)
+     * @param string $path     Specify the path where the cookie should be available. (Also in sub-paths)
+     * @param string $domain   Specify the domain where the cookie should be available. (Set ".example.org" to allow sub-domains)
+     * @param bool   $secure   If "true" cookie is only set if connection is HTTPS. Default is an auto detection.
+     * @param bool   $httpOnly If "true" cookie is accessible only via HTTP.
+     *                         Set to "false" to allow access for JavaScript. (Possible XSS security leak)
+     * @return bool Returns "true" if the cookie is successfully set.
      */
-    public function tableCleanup($maxInactiveMinutes = 30)
+    public static function setCookie($name, $value = '', $expire = 0, $path = '', $domain = '', $secure = null, $httpOnly = true)
```
Comment (thomas-rcv) [inline adm_program/system/classes/session.php]: We should also initialize $path, $domain and $value as undefined NULL.

File: adm_program/system/classes/session.php [hunk 2]
```diff
@@ -338,29 +291,19 @@ public function refreshSession()
...
         }
 
-        // if AutoLogin is set then refresh the auto_login_id for security reasons
-        if($this->getValue('ses_usr_id') > 0)
+        // session in database could be deleted if user was some time inactive and another user
+        // clears the table. Therefore we must reset the user id
+        if ($this->mAutoLogin instanceof \AutoLogin)
         {
-            if ($this->mAutoLogin instanceof \AutoLogin)
+            if((int) $this->getValue('ses_usr_id') === 0)
```
Comment (thomas-rcv) [inline adm_program/system/classes/session.php]: You could add the whitespace after "if"

File: adm_program/system/login_check.php [hunk 3]
```diff
@@ -109,24 +109,32 @@ function initLoginParams($prefix = '')
...
+        if (!array_key_exists($gCookiePraefix.'_PHP_SESSION_ID', $_COOKIE))
+        {
+            $gMessage->show($gL10n->get('SYS_COOKIE_NOT_SET', DOMAIN));
+            // => EXIT
+        }
 
-        // bei einer Beta-Version noch einen Hinweis ausgeben !
-        if(ADMIDIO_VERSION_BETA > 0 && !$gDebug)
+        // remove login page from navigation stack
+        if(strpos($gNavigation->getUrl(), '/login.php') > 0)
```
Comment (thomas-rcv) [inline adm_program/system/login_check.php]: whitespace

**Comment by thomas-rcv** [comment]
Reviewed. 
Small comments to codestyle as review comment.

File: adm_program/system/classes/session.php (hunk 2 above)
Comment (ximex) [inline adm_program/system/classes/session.php]: This codestyle isn't defined. But i would also be happy with this codestyle ;-)

File: adm_program/system/login_check.php (hunk 3 above)
Comment (ximex) [inline adm_program/system/login_check.php]: see above

File: adm_program/system/classes/session.php (hunk 1 above)
Comment (Fasse) [inline adm_program/system/classes/session.php]: that code was from ximex. So I don't want to change the default values.

File: adm_program/system/login_check.php (hunk 3 above)
Comment (Fasse) [inline adm_program/system/login_check.php]: We can set whitespaces later through ximex script.

**Pull request merged**